claripy/backends/__init__.py

```diff
@@ -600,6 +604,28 @@ def _max(self, expr, extra_constraints=(), solver=None, model_callback=None): #p
         """
         raise BackendError("backend doesn't support max()")
 
+    def check_satisfiability(self, extra_constraints=(), solver=None, model_callback=None):
```

Comment:
The `check_satisfiability` has the same purpose as `satisfiable` but instead of returning True/False it returns the string 'SAT', 'UNSAT' or 'UNKNOWN'. We implemented this function specifically for the portfolio solver since we need to be able to differentiate between 'UNSAT' and 'UNKNOWN'.

That function only calls `satisfiable` if the solver calling it does not have a specialized version of the function `_check_satisfiability`. For example all the solvers for the smtlib backend have that specialized function defined in `/claripy/backend/s/backend_smtlib_solver/__init__.py` at line 136. If you look at that function it doesn't call the function `satisfiable`.